backend/elasticsearch-client/common/src/main/java/org/hibernate/search/backend/elasticsearch/client/common/gson/spi/GsonProvider.java

```diff
@@ -21,24 +17,18 @@
 public final class GsonProvider {
 
 	/*
-	 * See https://github.com/google/gson/issues/764.
-	 *
-	 * Only composite type adapters (referring to another type adapter)
-	 * should be affected by this bug, so we'll list the corresponding types here.
-	 * Maybe it's even narrower, and only type adapters that indirectly refer to themselves are affected,
-	 * but I'm not entirely sure about that.
-	 *
-	 * Note we only need to list "root" types:
-	 * all types they refer to will also have their type adapter initialized.
+	 * https://github.com/google/gson/issues/764 is supposedly fixed.
+	 * Hence, we may not need the workaround anymore.
+	 * This version will be used in the test so we'll notice if things are still broken, for the main code
+	 * the workaround is in place through the GsonProviderHelper.
```

Comment:
This GSON issue was fixed, but I kept the workaround for the main code "just in case".